### ISSUE TITLE
memory error with pickle in _load_models()

### ISSUE THREAD

**Issue by qqliang** [automl/auto-sklearn#474]
I run auto-sklearn with 4 hours time budgets, when finished, it reports a memory error in automl.py -> _fit() -> self._load_models()，because it use pickle to load model file. How can I tackle this error?

**Comment by mfeurer** [comment]
We fixed this in the development branch (this was reported before as https://github.com/automl/auto-sklearn/issues/444). Therefore, you can either wait until we create a new release or use the development branch in the meantime.

**Comment by ushmalramesh** [comment]
I'm getting memory error with pickle in load_models while running auto-sklearn in parallel. The stack trace is as follows:

Traceback (most recent call last):
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/ushmal/PycharmProjects/automl/automl_parallel_sparse.py", line 72, in spawn_classifier
    automl.fit(X_train, y_train, dataset_name=dataset_name)
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/estimators.py", line 664, in fit
    dataset_name=dataset_name,
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/estimators.py", line 337, in fit
    self._automl[0].fit(**kwargs)
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 996, in fit
    load_models=load_models,
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 208, in fit
    only_return_configuration_space=only_return_configuration_space,
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/automl.py", line 370, in _fit
    data_manager_path = self._backend.save_datamanager(datamanager)
  File "/home/ushmal/anaconda3/envs/py36/lib/python3.6/site-packages/autosklearn/util/backend.py", line 318, in save_datamanager
    pickle.dump(datamanager, fh, -1)
MemoryError

**Comment by mfeurer** [comment]
According to your traceback, the memory error happens in the beginning of the fit method. Could you please open a new issue? Please also post the size of the dataset (shape, NNZ) and your amount of RAM?

**Comment by ushmalramesh** [comment]
opened issue #662
